src/gui/pedalboard/cables/CableView.cpp

```diff
@@ -31,45 +31,46 @@ void CableView::paint (Graphics& g)
         }
     }
 
-    g.setColour (cableColour.brighter (0.1f));
-    for (auto* cable : cables)
+    if (cableMouse != nullptr)
     {
-        auto* startEditor = board->findEditorForProcessor (cable->startProc);
-        jassert (startEditor != nullptr);
-
-        auto startPortLocation = CableViewPortLocationHelper::getPortLocation ({ startEditor, cable->startIdx, false });
-
-        if (cable->endProc != nullptr)
+        auto mousePos = cableMouse->getPosition();
+        const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cables.getLast()->startProc);
+        if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
         {
-            auto* endEditor = board->findEditorForProcessor (cable->endProc);
-            jassert (endEditor != nullptr);
+            auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
+            drawCablePortGlow (g, endPortLocation, scaleFactor);
+        }
+    }
+}
 
-            auto endPortLocation = CableViewPortLocationHelper::getPortLocation ({ endEditor, cable->endIdx, true });
+void CableView::resized()
+{
+    for (auto* cable : cables)
+    {
+        cable->setBounds (getLocalBounds());
+    }
+}
 
-            auto startColour = startEditor->getColour();
-            auto endColour = endEditor->getColour();
-            auto levelDB = cable->endProc->getInputLevelDB (cable->endIdx);
-            drawCable (g, startPortLocation.toFloat(), startColour, endPortLocation.toFloat(), endColour, scaleFactor, levelDB);
-        }
-        else if (connectionHelper->cableMouse != nullptr)
+void CableView::mouseDown (const MouseEvent& e)
+{
+    if (e.eventComponent == nullptr)
+        return; // not a valid mouse event
+
+    if (e.mods.isPopupMenu())
```

Comment:
At the moment this won't work on iOS, but that's okay. I can try to implement it sometime next week, or if you're feeling adventurous, you could try to do it using the `chowdsp::PopupMenuHelper`, like in the `BoardComponent`.

Reply:
I modified this with the newest changes. If it still won't work with iOS let me know and I might need some more background on what would need to happen for it to work with iOS. I would be interested to learn what the difference need to be!

Reply:
Yeah, so basically instead of using `e.mods.isPopupMenu()` in the `mouseDown()` implementation, you would have to add `chowdsp::PopupMenuHelper` [as a member variable](https://github.com/Chowdhury-DSP/BYOD/blob/main/src/gui/pedalboard/BoardComponent.h#L50), and then configure it [like this](https://github.com/Chowdhury-DSP/BYOD/blob/main/src/gui/pedalboard/BoardComponent.cpp#L63). I won't be able to test on iOS until sometime next week, but if you're able to get it working that way on desktop, I'm guessing it will work fine on iOS as well.

The reason for doing it this way is that there's no way to "right-click" on iOS, so we can configure the plugin to treat a "long-press" as a right-click, which requires some extra logic that is implemented in the helper.

Reply:
Good idea, I implemented a `popupMenu` with a callback in the `Cable` class. I thought about trying to implement the `popupMenu` in `Cableview`, but that would have required some more restructuring, to pass the clicked cable...

Let me know if you think this should work for IOS or if there could be a better way to organize it.

Reply:
Yeah, your implementation looks great! I'll see if I can give it a go on iOS later this week, but that shouldn't block us from merging this PR.